test/BaseUnit.GSquared.t.sol

```diff
@@ -0,0 +1,151 @@
+import "forge-std/Test.sol";
+import "forge-std/Vm.sol";
+import "forge-std/interfaces/IERC20.sol";
+import "./utils/utils.sol";
+import "../contracts/interfaces/IOracle.sol";
+import "../contracts/GRouter.sol";
+import "../contracts/GVault.sol";
+import "../contracts/GTranche.sol";
+import "../contracts/pnl/PnLFixedRate.sol";
+import "../contracts/mocks/MockERC20.sol";
+import "../contracts/mocks/Mock3CRV.sol";
+import "../contracts/mocks/MockDAI.sol";
+import "../contracts/mocks/MockUSDC.sol";
+import "../contracts/mocks/MockUSDT.sol";
+import "../contracts/mocks/MockUSDT.sol";
+import "../contracts/mocks/MockStrategy.sol";
+import "../contracts/mocks/MockCurveOracle.sol";
+import "../contracts/mocks/MockThreePoolCurve.sol";
+import "../contracts/solmate/src/utils/SafeTransferLib.sol";
+import {TokenCalculations} from "../contracts/common/TokenCalculations.sol";
+import "../contracts/solmate/src/utils/CREATE3.sol";
+
+contract BaseUnitFixture is Test {
```

Comment:
@kitty-the-kat most important changes are here, as this is fixture with all mocked externalities